src/main/java/com/example/HelloWorld/controller/HelloWorldController.java

```diff
@@ -35,12 +35,17 @@ public String greet(
       @RequestParam(name = "latest", required = false) boolean latest) {
 
     if (shout && reverse)
-      return new StringBuilder(helloWorldService.greetings(subject, latest, stack)).reverse().toString().toUpperCase(Locale.ROOT);
+      return new StringBuilder(helloWorldService.greetings(subject, latest, stack))
+          .reverse()
+          .toString()
+          .toUpperCase(Locale.ROOT);
 
     if (shout) return helloWorldService.greetings(subject, latest, stack).toUpperCase(Locale.ROOT);
```

Comment:
`helloWorldService.greetings(subject, latest, stack)` you call this multiple times, either externalize it or store the result and call it only once.